extensions/lifecycle/lifecycleProducer/LifecycleProducer.js

```diff
@@ -0,0 +1,256 @@
+        const str = JSON.stringify(entry);
+        this._bucketProducer.send([str], cb);
+    }
+
+    /**
+     * Get the state variables of the current instance.
+     * @return {Object} Object containing the state variables
+     */
+    getStateVars() {
+        return {
+            s3: this._s3Client,
+            sendBucketEntry: this.sendBucketEntry,
+            sendObjectEntry: this.sendObjectEntry,
+            enabledRules: this._lcConfig.rules,
+            log: this._log,
+        };
+    }
+
+    /**
+     * Determine whether the given config should be processed.
+     * @param {Object} config - The bucket lifecycle configuration
+     * @return {Boolean} Whether the config should be processed
+     */
+    _shouldProcessConfig(config) {
+        if (config.Rules.length === 0) {
+            this._log.debug('bucket lifecycle config has no rules to process', {
+                config,
+            });
+            return false;
+        }
+        const { rules } = this._lcConfig;
+        // Check if backbeat config has a lifecycle rule enabled for processing.
+        const enabled = Object.keys(rules).some(rule => rules[rule].enabled);
+        if (!enabled) {
+            this.log.debug('no lifecycle rules enabled in backbeat config');
+        }
+        return enabled;
+    }
+
+    /**
+     * Process the given bucket entry, get the bucket's lifecycle configuration,
+     * and schedule a task with the lifecycle configuration rules, if
+     * applicable.
+     * @param {Object} entry - The kafka entry containing the information for
+     * performing a listing of the bucket's objects
+     * @param {Object} entry.value - The value of the entry object
+     * @param {String} entry.value.bucket - The bucket which has objects to
+     * perform lifecycle actions on
+     * @param {String} entry.value.owner - The canonical ID of the bucket owner
+     * @param {String} [entry.value.prefix] - An object prefix
+     * @param {String} [entry.value.keyMarker] - KeyMarker to list objects from
+     * @param {String} [entry.value.versionIdMarker] - VersionIdMarker to list
+     * versioned objects from
+     * @param {String} [entry.value.continuationToken] - ContinuationToken
+     * indicating that the list is being continued on this bucket
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    _processBucketEntry(entry, cb) {
+        const { error, value } = safeJsonParse(entry.value);
+        if (error) {
+            this._log.error('could not parse bucket entry', { error });
+            return process.nextTick(() => cb(error));
+        }
+        return this._s3Client.getBucketLifecycleConfiguration({
+            Bucket: value.bucket,
+        }, (err, config) => {
+            if (err) {
+                this._log.error('error getting bucket lifecycle configuration',
+                    { error: err });
+                return process.nextTick(() => cb(err));
```

Comment:
since we're already in an async context (answer from `getBucketLifecycleConfiguration) we can get rid of the `process.nextTick`

Reply:
Updated.